src/lib/reasoners/theory.ml

```diff
@@ -996,22 +888,35 @@ module Main_Default : S = struct
   let get_case_split_env t = t.gamma_finite
 
   let compute_concrete_model env =
-    let { gamma_finite; assumed_set; objectives; _ }, _ =
-      do_case_split_aux env ~for_model:true in
-    CC_X.extract_concrete_model
-      ~prop_model:assumed_set
-      ~optimized_splits:objectives
-      gamma_finite
+    if Options.get_interpretation () then
+      let { gamma_finite; assumed_set; objectives; _ }, _ =
+        do_case_split_aux env ~for_model:true
+      in
+      Some (lazy (
+          CC_X.extract_concrete_model
+            ~prop_model:assumed_set
+            gamma_finite
+        ), objectives)
+    else
+      None
 
   let assume_th_elt t th_elt dep =
     { t with gamma = CC_X.assume_th_elt t.gamma th_elt dep }
 
   let get_assumed env = env.assumed_set
 
-  let reinit_cpt () =
-    Debug.reinit_cpt ()
+  let optimize env ~to_max f =
+    let obj = Objective.Function.mk ~to_max f in
+    let objectives =
+      Objective.Model.add obj Objective.Value.Unknown env.objectives
+    in
+    let env = reset_case_split_env env in
```

Comment:
There is a comment in `reset_case_split_env` that states: "be sure it's always correct when this function is called" (wrt the `cs_pending_facts`), but now we can call `reset_case_split_env` from a public function. Are we sure it's always correct?

(I think it is, because from my understanding setting `cs_pending_facts` to `[]` is always correct if we reset `gamma_finite` to `gamma`)

Reply:
Actually, I'm not even sure we need to invoke `reset_case_split` here. If it's the first time we register the objective `f`,  we may have propagate case split for `f` because `f` was involved in some assertions. But the objective value of `f` is `Unknown` and we will optimize it anyway.
If we have optimize some subexpressions of `f` here, it means their objectives have a higher priority and there is no reason to throw away these choices.

Reply:
Don't regular (non-optimizing) splits still need to be reset?

For instance if we assert ` 0 ≤ x ≤ 10`, there is currently no optimization under way so we pick `x = 0` as a regular split, then we do `(maximize x)`, we need to undo the `x = 0` choice.